aiida/orm/entities.py

```diff
@@ -118,15 +134,15 @@ def get(self, **filters):
 
         :return: the entry
         """
-        res = self.find(filters=filters)
-        if not res:
+        res = self.query(filters=filters, limit=1)
+        if res.count() == 0:
```

Comment:
Getting the count twice, here and two lines below, will most likely not be very efficient, unless we cache the result under the hood, which I doubt. So unless you can verify that the database is called once, maybe it is best to call `count` once, if you want to use that new interface

Reply:
Sure.

Looking at this method (`get`), I guess it's point has been superseded by `find`? It's point now is to return a single entity object through a `QueryBuilder` filter. However, the rigidity of this is not properly implemented I guess - and does the exception message for the `res.count() > 1` even return something meaningful? `id` is not a property of the `Collection` I guess, but rather of `Entity`...
It would make sense to simply remove `get` from `Collection` and have `Entity.get()` create a specific filter to be used with `find`, right? Or?

Reply:
> I guess it's point has been superseded by `find`?

No. The intention of `get` is to get a single specific entity of the collection. It should be uniquely resolvable, i.e. if none or more than one are found and exception should be raised. This should therefore also be a method on the collection and not the entity class itself.